### PR TITLE
FEATURE: Improve remove unused assets filter

### PR DESCRIPTION
**Pull request by daniellienert**
Improves the media:removeunused command

- Adds the option "onlyTags", which can be given as comma-separated list of tag labels to filter the assets that should be checked.
- Adds a flag --limit to limit the result of unused assets displayed and removed for a single run.

### CODE REVIEW

File: Neos.Media/Classes/Command/MediaCommandController.php [hunk 1]
```diff
@@ -164,35 +168,57 @@ public function removeUnusedCommand(string $assetSource = '', bool $quiet = fals
...
+            if (!($asset instanceof Asset)) {
+                continue;
+            }
+            if (!$asset instanceof AssetSourceAwareInterface) {
+                continue;
+            }
+            if ($filterByAssetSourceIdentifier !== '' && $asset->getAssetSourceIdentifier() !== $filterByAssetSourceIdentifier) {
+                continue;
+            }
+            if (trim($onlyTags) && !($assetTagsMatchFilterTags($asset->getTags(), $onlyTags))) {
```
Comment (bwaidelich) [inline Neos.Media/Classes/Command/MediaCommandController.php]: `trim($onlyTags)` is a weak comparison. Why not just `$onlyTags !== ''` or make it `null` by default?

**Comment by daniellienert** [comment]
@bwaidelich: Changed it

File: Neos.Media/Classes/Command/MediaCommandController.php [hunk 2]
```diff
@@ -164,35 +168,57 @@ public function removeUnusedCommand(string $assetSource = '', bool $quiet = fals
...
-                $unusedAssetCount++;
-                $unusedAssetsTotalSize += $asset->getResource()->getFileSize();
-            }
             !$quiet && $this->output->progressAdvance(1);
+
+            if ($unusedAssetCount === $limit) {
+                break;
+            }
+
+            if (!($asset instanceof Asset)) {
```
Comment (bwaidelich) [inline Neos.Media/Classes/Command/MediaCommandController.php]: &lt;nitpick&gt;in line 193 you do the same check without additional brackets and they are not needed indeed&lt;/nitpick&gt;

File: Neos.Media/Classes/Command/MediaCommandController.php [hunk 3]
```diff
@@ -164,35 +168,57 @@ public function removeUnusedCommand(string $assetSource = '', bool $quiet = fals
...
+                continue;
+            }
+            if (!$asset instanceof AssetSourceAwareInterface) {
+                continue;
+            }
+            if ($filterByAssetSourceIdentifier !== '' && $asset->getAssetSourceIdentifier() !== $filterByAssetSourceIdentifier) {
+                continue;
+            }
+            if ($onlyTags === '' && !($assetTagsMatchFilterTags($asset->getTags(), $onlyTags))) {
```
Comment (bwaidelich) [inline Neos.Media/Classes/Command/MediaCommandController.php]: Shouldn't that be `$onlyTags !== ''`?!

File: Neos.Media/Classes/Command/MediaCommandController.php (hunk 2 above)
Comment (daniellienert) [reply Neos.Media/Classes/Command/MediaCommandController.php]: Fixed

File: Neos.Media/Classes/Command/MediaCommandController.php (hunk 3 above)
Comment (daniellienert) [reply Neos.Media/Classes/Command/MediaCommandController.php]: Gnah, sorry

**Comment by daniellienert** [comment]
@bwaidelich: Would you please have another look?

**Comment by robertlemke** [comment]
For the record, the `--limit` flag introduced a regression, fixed here: https://github.com/neos/neos-development-collection/pull/2289

**Pull request merged**